src/app/context/AccountContext.tsx

```diff
@@ -93,46 +129,45 @@ export function AccountProvider({ children }: { children: React.ReactNode }) {
     setAccountBalance(balance);
   };
 
-  const fetchAccountInfo = async (options?: { accountId?: string }) => {
-    const id = options?.accountId || account?.id;
-    if (!id) return;
-
-    const callback = (accountRes: AccountInfo) => {
-      setAccount(accountRes);
-      updateAccountBalance(accountRes.balance, accountRes.currency);
-    };
-
-    const accountInfo = await api.swr.getAccountInfo(id, callback);
+  const handleGetStatus = (status: StatusRes) => {
+    setStatus(status);
+    const onWelcomePage = window.location.pathname.indexOf("welcome.html") >= 0;
+    if (!status.configured && !onWelcomePage) {
+      utils.openPage("welcome.html");
+      window.close();
+    } else if (status.unlocked) {
+      if (status.configured && onWelcomePage) {
+        utils.redirectPage("options.html");
+      }
+      setAccountId(status.currentAccountId);
+    } else {
+      setAccount(null);
+    }
+  };
 
-    return { ...accountInfo, fiatBalance, accountBalance };
+  const setAccountData = (
+    id: string | undefined,
+    accountInfoRes: AccountInfoRes | undefined
+  ) => {
+    if (id && accountInfoRes) {
+      const accountInfo = buildAccountInfo(id, accountInfoRes);
+      setAccount(accountInfo);
+      updateAccountBalance(accountInfo.balance, accountInfo.currency);
+    }
   };
 
+  useEffect(() => {
+    setAccountData(account?.id, accountInfoRes);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
can you fix the warning rather than disabling it?

Reply:
I would need to add setAccountData and apply useCallback() - I didn't really get it working and seems like this makes it overly complicated. Is it a problem though? This exception is used in many other effects!?

Reply:
It might work fine. It is just a code smell. I'm not sure how important it is to fix for now.